### PR TITLE
Add new regex for `concretization_error`

### PR DESCRIPTION
**Pull request by mvandenburgh**
Examples:
https://gitlab.spack.io/spack/spack/-/jobs/13067136 
https://gitlab.spack.io/spack/spack/-/jobs/13067292

### CODE REVIEW

File: analytics/analytics/core/job_failure_classifier/taxonomy.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ taxonomy:
         - "failed to concretize .+ for the following reasons"
         - "variant .+ not found in package"
         - "trying to set variant .+ in package .+, but the package has no such variant"
+        - "UnknownVariantError: No such variant .+ for spec"
```
Comment (scottwittenburg) [inline analytics/analytics/core/job_failure_classifier/taxonomy.yaml:30]: Just wondering if reducing this to `"No such variant"` would catch a wider variety of concretization issues without risking false positives.

Or maybe adding `"UnknownVariantError"` and `"No such variant"` as two separate regexes or perhaps a single regex with a `|`?

I'm actually surprised we don't have such an expression already, I'll check.

Comment (scottwittenburg) [reply analytics/analytics/core/job_failure_classifier/taxonomy.yaml:30]: I guess they're all right there under `concretization_error`.  Maybe the ones above correspond to older variant-related concretization errors, or maybe they're still valid, not sure.

Comment (mvandenburgh) [reply analytics/analytics/core/job_failure_classifier/taxonomy.yaml]: False negatives seem to be a larger issue than false positives when it comes to concretization errors, so splitting them up is fine with me :+1:

**Review by scottwittenburg — APPROVED** [review]
Looks good, thanks @mvandenburgh!

**Pull request merged**